packages/block-editor/src/store/selectors.js

```diff
@@ -1798,3 +1798,45 @@ export function isBlockHighlighted( state, clientId ) {
 export function areInnerBlocksControlled( state, clientId ) {
 	return !! state.blocks.controlledInnerBlocks[ clientId ];
 }
+
+/**
+ * Returns an array of clientIds for hovered blocks set in full site editing.
+ * Block Ids set in FSE are ordered from child to parent.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {string[]} Array of clientIds for hovered blocks.
+ */
+export function __experimentalGetHoveredBlocks( state ) {
+	return state.hoveredBlocks.hoveredBlockIds;
+}
+
+/**
+ * Returns the most interior hovered block of the given blockName (such as 'core/group').
+ *
+ * @param {Object} state     Global application state.
+ * @param {string} blockName Name of block type to return.
+ *
+ * @return {Object} Parsed block object.
+ */
+export const __experimentalGetHoveredBlockByBlockName = createSelector(
+	( state, blockName ) => {
+		const matchingBlockClientId = state.hoveredBlocks.hoveredBlockIds.find(
+			( blockId ) => getBlockName( state, blockId ) === blockName
+		);
+		return getBlock( state, matchingBlockClientId );
+	},
+	( state ) => [ state.hoveredBlocks.hoveredBlockIds, state.blocks.cache ]
```

Comment:
I didn't realize this (so my bad for the recommendation), but I now realize that `state.blocks.cache` won't work. It's meant to be like `state.blocks.cache[ clientId ]`, which contains an object which only changes reference when the block is updated. Given that we don't have a clientId here, I guess it doesn't make sense to memoize on that.

But... If we only memoize on hoveredBlockIds, it would return stale block attributes if the block attributes have updated but the hovered state has not updated. 

So maybe we should only return the client ID, and then only memoize on `state.hoveredBlocks.hoveredBlockIds`? (I don't think block name can change at runtime)

Hm.

Reply:
So I just did these changes locally, then started wondering more about:
> I now realize that `state.blocks.cache` won't work.

At first I would think this is true because we are checking the same object.  But looking at the cache reducer, every action that changes the cache, creates a new cache object and old values are spread into it.  So any time the cache is updated, `state.blocks.cache` changes to a new object, and it should work as a dependency here? 🤔

Reply:
that's true. I guess that would refresh the selector whenever any block changes which makes sure the returned block is up to date :)

I still like the idea of just returning the clientId and letting it just cache on hovereBlocks. Feels more "pure" that way, I guess.

Either way, it's not a big deal IMO.

Reply:
true!  and I have that all ready to go, might as well. 😄